### ISSUE TITLE
Cox partial likelihood doesn't seem to index over all events

### ISSUE THREAD

**Issue by lyyraaa** [Novartis/torchsurv#117]
In the partial likelihood functions in https://github.com/Novartis/torchsurv/blob/main/src/torchsurv/loss/cox.py :
- _partial_likelihood_cox
- _partial_likelihood_efron
- _partial_likelihood_breslow  

The array of partial log likelihood contributions for each event is indexed using a binary tensor of event status `event_sorted` (or `include`)
e.g. `_partial_likelihood_cox` has ` return (log_hz_sorted - log_denominator)[event_sorted]`

I believe this is meant to perform the outer summation of this formula:
<img width="321" height="77" alt="Image" src="https://github.com/user-attachments/assets/7cb3f0ec-5058-425b-89c7-41e9f20162e1" />
(https://opensource.nibr.com/torchsurv/_autosummary/torchsurv.loss.cox.html)
 by returning only values associated with an event (e.g. `event_sorted == 1`)
However, it instead repeatedly indexes just the first and second values:
```python
def _partial_likelihood_cox(
    log_hz_sorted: torch.Tensor,
    event_sorted: torch.Tensor,
) -> torch.Tensor:
    """Calculate the partial log likelihood for the Cox proportional hazards model
    in the absence of ties in event time.
    """
    log_hz_flipped = log_hz_sorted.flip(0)
    log_denominator = torch.logcumsumexp(log_hz_flipped, dim=0).flip(0)
    return (log_hz_sorted - log_denominator)[event_sorted]

dummy_log_hz = torch.tensor([-1, 1, 0, 0.5, 0.5, 0.5])
dummy_event = torch.tensor([0, 1, 0, 1, 1, 1])

print(_partial_likelihood_cox(dummy_log_hz, dummy_event))
```
`> tensor([-3.2008, -1.1592, -3.2008, -1.1592, -1.1592, -1.1592])`

If my interpretation of how these functions are meant to work is correct, then this could be fixed by using the indexes of events in `event_sorted`:
```python
return (log_hz_sorted - log_denominator)[torch.where(event_sorted==1)]
```
`> tensor([-1.1592, -1.0986, -0.6931,  0.0000])`
